redisbench_admin/run_async/run_async.py

```diff
@@ -0,0 +1,460 @@
+    if module_check_status is False:
+        if webhook_notifications_active:
+            failure_reason = error_message
+            generate_failure_notification(
+                webhook_client_slack,
+                ci_job_name,
+                ci_job_link,
+                failure_reason,
+                tf.tf_github_org,
+                tf.tf_github_repo,
+                tf.tf_github_branch,
+                None,
+            )
+        exit(1)
+
+    # log properties for terraform
+    tf.common_properties_log(private_key)
+
+    ssh_pem_check(EC2_PRIVATE_PEM, private_key)
+
+    benchmark = BenchmarkClass()
+
+    benchmark.prepare_benchmark_definitions(args)
+
+    return_code = 0
+    if benchmark.benchmark_defs_result is False:
+        return_code = 1
+        if args.fail_fast:
+            failure_reason = "Detected errors while preparing benchmark definitions"
+            logging.critical("{}. Exiting right away!".format(failure_reason))
+            if webhook_notifications_active:
+                generate_failure_notification(
+                    webhook_client_slack,
+                    ci_job_name,
+                    ci_job_link,
+                    failure_reason,
+                    tf.tf_github_org,
+                    tf.tf_github_repo,
+                    tf.tf_github_branch,
+                    None,
+                )
+
+            exit(return_code)
+
+    (
+        _,
+        _,
+        _,
+        tsname_project_total_failures,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+    ) = get_overall_dashboard_keynames(
+        tf.tf_github_org, tf.tf_github_repo, tf.tf_triggering_env
+    )
+
+    benchmark.populate_remote_envs_timeout()
+
+    for remote_id, termination_timeout_secs in benchmark.remote_envs_timeout.items():
+        logging.info(
+            "Using a timeout of {} seconds for remote setup: {}".format(
+                termination_timeout_secs, remote_id
+            )
+        )
+
+    # we have a map of test-type, dataset-name, topology, test-name
+    benchmark.benchmark_runs_plan = define_benchmark_plan(
+        benchmark.benchmark_definitions, benchmark.default_specs
+    )
+
+    # create service file
+    renderServiceFile(
+        access_key=EC2_ACCESS_KEY,
+        secret_key=EC2_SECRET_KEY,
+        region=EC2_REGION,
+        gh_token=os.getenv("GH_TOKEN", None),
+        job_name=os.getenv("CIRCLE_JOB", None),
+        args=args,
+        argv=argv,
+    )
+
+    # create run.py file for running redisbench-cli
+    renderRunFile()
+    savePemFile(EC2_PRIVATE_PEM)
+
+    # copy module file
+    if len(args.module_path) != 0:
+        copy(args.module_path[0], os.getcwd())
+
+    # zip all
+    archive_name = tar_files()
+
+    # create runner
+    tf.async_runner_setup()
+
+    # push archive to runner
+
+    copy_file_to_remote_setup(
+        tf.runner_public_ip,
+        "ubuntu",
```

Comment:
should the username be a arg? assuming in the future we might be dealing with different os's and different default users